node/src/chain.rs

```diff
@@ -332,7 +332,7 @@ where
                     // on the URL)
                     Ok(Err(e)) | Err(e) => {
                         error!(logger, "Connection to provider failed. Not using this provider";
-                                       "error" =>  e.to_string());
+                                       "error" =>  format!("{:#}", e));
```

Comment:
Was this done for debugging purposes? It's probably wise to keep it so it's easier to understand the problem in the future

Reply:
imo we should try to remember to always format errors like this, in case they are anyhow errors.

Reply:
Yes, but it led to nothing interesting. 

However, this is the best way to print `anyhow::Error`. If the anyhow error has multiple causes, they will all be printed when using the `:#` formatting directive while `to_string()` or `{}` only prints the outermost error, see https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations

Reply:
Ahah Leo was faster than me :D At least I gave a good reference link!